internal/pkg/template/templates/environment/partials/custom-resources.yml

```diff
@@ -44,5 +44,10 @@ CustomDomainAction:
     Aliases: !Ref Aliases
     AppDNSRole: !Ref AppDNSDelegationRole
     DomainName: !Ref AppDNSName
-    LoadBalancerDNS: !GetAtt PublicLoadBalancer.DNSName
-    LoadBalancerHostedZone: !GetAtt PublicLoadBalancer.CanonicalHostedZoneID
\ No newline at end of file
+    {{- if .CDNConfig}}
+    PublicAccessDNS: !GetAtt CloudFrontDistribution.DomainName
+    PublicAccessHostedZone: Z2FDTNDATAQYW2 # See https://go.aws/3cPhvlX
+    {{- else}}
+    PublicAccessDNS: !GetAtt PublicLoadBalancer.DNSName
+    PublicAccessHostedZone: !GetAtt PublicLoadBalancer.CanonicalHostedZoneID
```

Comment:
This will re-trigger the CustomDomainAction for all existing customers that already have the action enabled. Are we certain that when we try to insert the records again it won't fail and the actions are idempotent?

Reply:
As far as I know, the only changes the lambda function may have on subsequent calls with the same input have to do with whether the resource does not respond in time. Otherwise, it will write the same A-record so it shouldn't ever cause an issue for the users, nor should it cause down-time.